### PR TITLE
Bump Flink version to 1.16

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

Bump Flink version to  1.16. And remove workarounds that are not necessary after Flink 1.16.

## Brief change log

- Bump Flink version to 1.16.
- Set rest address and port with StreamExecutionEnvironment config.
- FlinkProcessor overwindow support first/last value

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by Sxnan** [comment]
@lindong28 Could you review this PR?

File: python/feathub/processors/flink/job_submitter/flink_session_cluster_job_submitter.py [hunk 1]
```diff
@@ -80,20 +80,21 @@ def submit(
             )
             return FlinkSessionClusterJob(None, self._executor)
 
-        native_flink_table = self.flink_processor.flink_table_builder.build(
-            features=features,
-            keys=keys,
-            start_datetime=start_datetime,
-            end_datetime=end_datetime,
-        )
+        with self.flink_processor.flink_table_builder.class_loader:
```
Comment (lindong28) [inline python/feathub/processors/flink/job_submitter/flink_session_cluster_job_submitter.py:83]: Do pyflink users always need to explicitly specify class loader in order to submit Flink jobs? If not, can you explain why we need it here?

File: python/feathub/processors/flink/table_builder/over_window_utils.py [hunk 2]
```diff
@@ -239,8 +239,10 @@ def _get_over_window_agg_select_expr(
         result = expr.sum
     elif agg_func == AggFunc.VALUE_COUNTS:
         result = native_flink_expr.call("value_counts", expr)
-    # TODO: FIRST_VALUE AND LAST_VALUE is supported after PyFlink 1.16 without
-    # PyFlink UDAF.
+    elif agg_func == AggFunc.FIRST_VALUE:
```
Comment (lindong28) [inline python/feathub/processors/flink/table_builder/over_window_utils.py:242]: Should we update `FlinkProcessorITTest` to enable the corresponding tests?

**Review by lindong28 — COMMENTED** [review]
Thanks for the PR! LGTM overall. Left two comments below.

File: python/feathub/processors/flink/table_builder/over_window_utils.py (hunk 2 above)
Comment (Sxnan) [reply python/feathub/processors/flink/table_builder/over_window_utils.py:242]: I updated the `FlinkProcessorITTest` to enable the corresponding tests.

File: python/feathub/processors/flink/job_submitter/flink_session_cluster_job_submitter.py (hunk 1 above)
Comment (Sxnan) [reply python/feathub/processors/flink/job_submitter/flink_session_cluster_job_submitter.py:83]: I added a comment and the related Flink JIRA ticket at the `__init__` method of `FlinkTableBuilder`.

**Comment by Sxnan** [comment]
@lindong28 Thanks for your review! I updated the PR accordingly. Could you please take another look?

**Comment by lindong28** [comment]
Thanks for the update. LGTM.

**Pull request closed without merging**